packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -98,17 +98,18 @@ export class Dropdown extends BaseComponent<IDropdownInternalProps, IDropdownSta
   public componentWillReceiveProps(newProps: IDropdownProps) {
     // In controlled component usage where selectedKey is provided, update the selectedIndex
     // state if the key or options change.
-    if (newProps.selectedKey !== undefined &&
+    if (this.props.multiSelect &&
+      newProps.selectedKeys !== undefined &&
+      (newProps.selectedKeys !== this.props.selectedKeys || newProps.options !== this.props.options)) {
+      this.setState({
```

Comment:
there's a lot of redundancy here; possible to trip down? Destructure? Would it be better to just store a selectedIndexes state even in the single scenario?

Reply:
I followed your suggestion, and added a truckload of new tests. However, there are two new ones (currently commented out) that seem to be failing due to some bug in the tests, as I've manually verified the scenarios. Not sure if you can help with the failing tests, or are willing to merge with what's passing.